### PR TITLE
vcontrold (new formula)

### PR DESCRIPTION
**Pull request by reitermarkus**
Tool for controlling and logging Viessmann boilers.

### CODE REVIEW

File: vcontrold.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+class Vcontrold < Formula
+  desc 'Tool for controlling and logging Viessmann boilers.'
+  homepage 'http://openv.wikispaces.com/'
```
Comment (DomT4) [inline vcontrold.rb]: This can be `https://`

File: vcontrold.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  head 'http://svn.code.sf.net/p/vcontrold/code/trunk/vcontrold/'
+
+  depends_on 'automake'
```
Comment (DomT4) [inline vcontrold.rb]: The autotool deps should be tagged with `=> :build`.

File: vcontrold.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  depends_on 'autoconf'
+  depends_on 'libxml2'
+  
+  option "client-only", 'Only install client.'
```
Comment (DomT4) [inline vcontrold.rb]: All options should start with `with` or `without`.

File: vcontrold.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+  def install
+
```
Comment (DomT4) [inline vcontrold.rb]: No need for whitespace here.

File: vcontrold.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  depends_on 'automake'
+  depends_on 'autoconf'
+  depends_on 'libxml2'
+  
+  option "client-only", 'Only install client.'
+
+  def install
+	  
+	if build.include? "client-only"
+	  system 'echo "sbin_PROGRAMS =" >> Makefile.am'
```
Comment (DomT4) [inline vcontrold.rb]: Is this officially supported upstream? I don't particularly want to hack up the `Makefile` to enable an option that upstream won't support.

File: vcontrold.rb [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+	  system 'echo "bin_PROGRAMS = vclient" >> Makefile.am'
+    end
+
+	system 'chmod +x auto-build.sh'
```
Comment (DomT4) [inline vcontrold.rb]: Don't need to call system here. `chmod 0755, auto-build.sh` is fine.

File: vcontrold.rb [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    system './auto-build.sh'
+
+    system "./configure", "--disable-silent-rules",
+                          "--prefix=#{prefix}"
+
+    system 'make', 'install'
+  end
+end
```
Comment (DomT4) [inline vcontrold.rb]: Need an empty new line at the end of the file.

File: vcontrold.rb [hunk 8]
```diff
@@ -0,0 +1,27 @@
+class Vcontrold < Formula
+  desc 'Tool for controlling and logging Viessmann boilers.'
```
Comment (DomT4) [inline vcontrold.rb]: Should be double quotes around everything.

**Comment by reitermarkus** [comment]
I tried `chmod 0755, auto-build.sh`, but it gave me this error:

    configure: error: cannot find install-sh, install.sh, or shtool in "." "./.." "./../.."
`

**Comment by DomT4** [comment]
It's working okay for me?

```ruby
    chmod 0755, "auto-build.sh"
    system "./auto-build.sh"
    system "./configure", "--disable-silent-rules", "--prefix=#{prefix}"
    system "make", "install"
```

```
==> ./auto-build.sh
configure.ac:13: installing './compile'
configure.ac:7: installing './install-sh'
configure.ac:7: installing './missing'
==> ./configure --disable-silent-rules --prefix=/usr/local/Cellar/vcontrold/HEAD
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... ./install-sh -c -d
checking for gawk... no
checking for mawk... no
checking for nawk... no
checking for awk... awk
checking whether make sets $(MAKE)... yes
```

**Comment by reitermarkus** [comment]
Ok, I didn't put `auto-build.sh` in parentheses, it's working now.

File: vcontrold.rb [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+  head "http://svn.code.sf.net/p/vcontrold/code/trunk/vcontrold/"
+
+  option "without-server", "Only install client."
+
+  depends_on "automake" => :build
+  depends_on "autoconf" => :build
+  depends_on "libxml2"
+
+  def install
+    if build.include? "without-server"
```
Comment (DomT4) [inline vcontrold.rb]: `if build.without? "server"`

File: vcontrold.rb [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+      system "echo 'sbin_PROGRAMS=' >> Makefile.am"
```
Comment (DomT4) [inline vcontrold.rb]: Likewise, would still like to know what the status is on supporting this upstream, and we should use an `inreplace`.

**Comment by reitermarkus** [comment]
I don't know if it is supported, but I also don't think it's going to change. But it is unnecessary to install the server tools on a client-only machine.

**Comment by reitermarkus** [comment]
Can this be merged now or do I have to remove the Makefile manipulation?

**Comment by DomT4** [comment]
This is currently blowing up the test-bot consistently, which is the hold-up. Doing a local audit raises this:

```
~> auditst vcontrold.rb
==> brew style vcontrold

1 file inspected, no offenses detected
Error: undefined method `downcase' for nil:NilClass
Please report this bug:
    https://git.io/brew-troubleshooting
/usr/local/Library/Homebrew/cmd/audit.rb:423:in `audit_specs'
/usr/local/Library/Homebrew/cmd/audit.rb:814:in `audit'
/usr/local/Library/Homebrew/cmd/audit.rb:53:in `block in audit'
/usr/local/Library/Homebrew/cmd/audit.rb:51:in `each'
/usr/local/Library/Homebrew/cmd/audit.rb:51:in `audit'
/usr/local/Library/brew.rb:140:in `<main>'
```

I would guess the error is being thrown from [this block](https://github.com/Homebrew/homebrew/blob/master/Library/Homebrew/cmd/audit.rb#L422-L453) but I haven't had a huge amount of time to look it over yet.

CC @Homebrew/owners

**Comment by reitermarkus** [comment]
This is my first formula, so I don't have much experience with fixing errors. Is there anything I can do to help solve this?

**Comment by DomT4** [comment]
Just need to fix the one last thing:

```
 * Description is too long. "name: desc" should be less than 80 characters (currently 81).
```

Ideally a `test do` block needs adding as well - If it's possible to test it somehow.

File: vcontrold.rb [hunk 11]
```diff
@@ -0,0 +1,26 @@
+class Vcontrold < Formula
+  desc "Tool for controlling and logging Viessmann boilers."
+  homepage "https://openv.wikispaces.com/vcontrold"
+  head "http://svn.code.sf.net/p/vcontrold/code/trunk/vcontrold/"
+
+  option "without-server", "Install client only."
```
Comment (mikemcquaid) [inline vcontrold.rb]: What's the advantage of this? Does it dramatically improve build time?

**Comment by reitermarkus** [comment]
Not really, as it is a really small program. I just added the option because the server command is essentially useless on a client machine.

**Comment by mikemcquaid** [comment]
@reitermarkus Let's just remove the option then, thanks!

**Comment by reitermarkus** [comment]
As you can see in the `test do`, the command never returns true, even if only help is displayed.

**Comment by DomT4** [comment]
You can do the same thing you're doing now with `assert_match /usage/, shell_output("#{bin}/vcontrold -h")` and it may work. If that doesn't exit cleanly we can handle that.

**Comment by reitermarkus** [comment]
Ok, i now added this: 

        assert_match /usage/, shell_output("#{sbin}/vcontrold --help")


`brew test vcontrold`:

    Testing homebrew/head-only/vcontrold
    ==> /usr/local/Cellar/vcontrold/HEAD/sbin/vcontrold --help
    Error: homebrew/head-only/vcontrold: failed
    <0> expected but was
    <1>.

**Comment by DomT4** [comment]
Try `assert_match /usage/, shell_output("#{sbin}/vcontrold --help", 1)`

**Comment by reitermarkus** [comment]
It still says the branch has failed checks, how can I see which checks have failed exactly?

Or can it still be merged with failed checks?

**Comment by DomT4** [comment]
Apologies, our test-bot crashed there. I've asked it to look again now.

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew-head-only/commit/8717dc91cde4f5f8fa2416ad0a4806f7f02db077. Thank you for your contribution to `head-only`, we appreciate it! :smiley_cat:

**Pull request closed without merging**